src-java/flowhs-topology/flowhs-storm-topology/src/main/java/org/openkilda/wfm/topology/flowhs/fsm/common/actions/haflow/BaseHaFlowPathRemovalAction.java

```diff
@@ -95,7 +95,7 @@ protected void removeRejectedPaths(Collection<PathId> subPathIds, Collection<Pat
         for (PathId haFlowPathId : haFlowPathIds) {
             haFlowPathRepository.remove(haFlowPathId)
                     .ifPresent(haFlowPath -> {
-                        HaFlowHistoryService.using(carrier).save(HaFlowHistory.withTaskId(correlationId)
+                        FlowHistoryService.using(carrier).save(HaFlowHistory.of(correlationId)
```

Comment:
Same here

Reply:
I created a local variable for the flow history service.